apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ids/GlobalIdGenerator.java

```diff
@@ -73,9 +73,19 @@ private long timestamp() {
             long currentTimeMillis = System.currentTimeMillis();
 
             if (currentTimeMillis < lastTimestamp) {
+                long overTime = lastTimestamp - currentTimeMillis;
+                if (overTime <= 5) {
+                    try {
+                        Thread.sleep(overTime);
```

Comment:
We can't afford this sleep.

Reply:
![image](https://user-images.githubusercontent.com/16626084/114339294-6908b280-9b87-11eb-98f0-d95c44c2e2f1.png)
Change `overTime` to 1ms.
Use `LockSupport.parkNanos` instead of `Thread.sleep`.
![image](https://user-images.githubusercontent.com/16626084/114339407-b08f3e80-9b87-11eb-9fef-1b6968ca9de8.png)

Actually, the `NTP` time offset(time shift) is very small in Linux servers.
`1ms` may cover most cases, while keeping good performance.
![image](https://user-images.githubusercontent.com/16626084/114340192-96566000-9b89-11eb-96cf-246cb04d1a46.png)

If ok, I will modify this PR.

Reply:
No, `1ms` is not acceptable. The agent can't block the user process, no matter in any case.

Reply:
APM is a low priority system, `1ms` is not in a good performance.

Reply:
> APM is a low priority system, `1ms` is not in a good performance.

This is a very good reason to avoid `sleep` indeed.
Inspired by your comments, there is an elegant solution without `sleep` or `random`.

General idea:
Use increment instead of `random`, which avoid conflict and improve performance
Old:
![image](https://user-images.githubusercontent.com/16626084/114383131-c6b8f100-9bbf-11eb-8a31-b6636f18d556.png)
New:
![image](https://user-images.githubusercontent.com/16626084/114383160-cf112c00-9bbf-11eb-8475-74fca1e26e30.png)

If ok, I will modify this PR.

Reply:
This could face concurrency status, `lastShiftValue++` could be the same in different threads.

Reply:
![image](https://user-images.githubusercontent.com/16626084/114402077-13a7c200-9bd6-11eb-8ba5-7d36111d0efe.png)
![image](https://user-images.githubusercontent.com/16626084/114402341-4f428c00-9bd6-11eb-9af8-f1e605bcd631.png)

`ThreadLocal` means every thread hold different `IDContext`.